### PR TITLE
ipc_service: Adding functions to allow for the teardown of an IPC instance

### PR DESCRIPTION
**Pull request by jacksonc-xlnx**
This builds upon my previous patch which allowed you to teardown an endpoint. This patch allows the whole ipc_service instance to be closed. It is intended to be used to tidy up in the case that an endpoint goes down or there is another failure in the ipc_service.

### CODE REVIEW

**Comment by carlescufi** [comment]
@carlocaione @hubertmis could you take a look please?

**Review by carlocaione — APPROVED** [review]

**Review by hubertmis — APPROVED** [review]

**Comment by jacksonc-xlnx** [comment]
@carlocaione Can this now be merged?

**Comment by carlocaione** [comment]
> @carlocaione Can this now be merged?

We are in code freeze now. This must wait for the release to be out before merging.

**Pull request merged**
